packages/core-interfaces/src/core-state/wallets.ts

```diff
@@ -55,6 +56,24 @@ export interface IWalletManager {
 
     findByUsername(username: string): IWallet;
 
+    findByBusinessName(name: string): IWallet;
+
+    findByBusinessWebsite(website: string): IWallet;
+
+    findByBusinessVat(vat: string): IWallet;
+
+    findByBusinessGithub(github: string): IWallet;
+
+    hasBusiness(businessAsset: Interfaces.IBusinessRegistrationAsset): boolean;
+
+    hasByBusinessName(name: string): boolean;
```

Comment:
I would only keep this on if you don't use any of the others `byBusinessWebsite`, `byBusinessVat`, `byBusinessGithub`, `hasBusiness` in your code.
It might be relevant to have an index by business name, but not necessary by all the other properties, so I would keep it simple for now (unless some of the others have some use for you).

Reply:
I will remove those methods, I made them only because I thought I would call them separately in transaction.
About the indexes i thought this would be the best way to to check all unique fields, but in the other hand I could index only name and loop through for all other unique fields.